Added documentation for `index_labels`. (#2465)
ManimCommunity/manim: docs/source/tutorials/using_text.rst

This makes transformations between similar text fragments easy
to write using :class:`~.TransformMatchingTex`.

Using ``index_labels`` to work with complicated strings
=======================================================

> You might sometimes be working with a very complicated ``MathTex`` mobject

Reviewer: Maybe turn this into a reference?

```suggestion
You might sometimes be working with a very complicated :class:`~.MathTex` mobject
```

that makes it difficult to work with its individual components. This is

> where a debugging function :class:`~.index_labels` is very useful.

Reviewer: ```suggestion
where the debugging function :func:`.index_labels` is very useful.
```

> `index_labels` shows the index of a mobject's submobjects, allowing you

Reviewer: This is to avoid starting the sentence with lower letters.

```suggestion
The method shows the index of a mobject's submobjects, allowing you
```

> to easily find the components of the mobject you'd like to change.

Reviewer: I think we more or less consistently avoid abbreviations like that.

```suggestion
to easily find the components of the mobject you would like to change.
```
